src/gpuarray_array.c

```diff
@@ -1160,7 +1159,7 @@ int GpuArray_fdump(FILE *fd, const GpuArray *a) {
     default:
       free(buf);
       fprintf(fd, "<unsupported data type %d>\n", a->typecode);
-      return GA_UNSUPPORTED_ERROR;
+      return error_set(ctx->err, GA_UNSUPPORTED_ERROR, "Unsupported data type for dump");
```

Comment:
Since this is a debugging help, I want it to be chatty when the data type is not supported.  

I've added the typecode to the error message.